sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,360 @@ def named_poly(n, f, K, name, x, polys):
+
+    Takes lists of (factor, conditions) pairs and returns minimal combinations that
+    satisfy the system of equations.
+    """
+
+    factor_dict = defaultdict(list)
+    for eq in factored_eqs:
+        for factor, conditions in eq:
+            factor_dict[factor].extend(conditions)
+
+    eq_factors = [{factor for factor, _ in eq} for eq in factored_eqs]
+
+    all_combinations = _find_minimal_factor_combinations(eq_factors)
+
+    minimal_combinations = []
+    for combo in all_combinations:
+        if not any(combo > other for other in all_combinations):
+            minimal_combinations.append(combo)
+
+    return [[(factor, factor_dict[factor]) for factor in combo]
+            for combo in minimal_combinations]
+
+
+def factor_system_cond(eqs, gens=None, **kwargs):
+    """Factor a system of equations and return factors with symbolic conditions.
+
+    Converts equations to polynomials, factors them, and returns factors with
+    Boolean conditions expressing when each factor is valid. Handles both
+    polynomial and transcendental equations.
+
+    Parameters
+    ==========
+
+    eqs : list
+        List of equations or expressions to be factored. Can include:
+        - Polynomial equations
+        - Transcendental equations
+        - Zero expressions
+        - Constants
+    gens : list, optional
+        List of generators (variables). If None, determined automatically
+        from the equations' free symbols
+    **kwargs : dict
+        Additional keyword arguments for factorization routines
+
+    Returns
+    =======
+
+    list of list of tuple
+        Each inner list represents a solution set.
+        Each tuple contains (factor, condition) where:
+        - factor is a symbolic expression (not a Poly)
+        - condition is either:
+            * sympy.true if no conditions apply
+            * Or(Ne(c1, 0), Ne(c2, 0), ...) for non-zero conditions
+
+    Examples
+    ========
+
+    >>> from sympy import symbols
+    >>> from sympy.polys.polytools import factor_system_cond
+    >>> x, a, b = symbols('x a b')
+    >>> factor_system_cond([(a - 1)*(x - 2), (b - 3)*(x - 2)], [x])
+    [[(x - 2, Ne(a - 1, 0) | Ne(b - 3, 0))]]
+
+    Notes
+    =====
+
+    - Returns [] if input list is empty
+    - Returns [[]] if all expressions are zero, indicating infinite solutions
+    - Returns [] if any non-zero constant is present, indicating no solution exists
+    - Handles transcendental equations that can be converted to polynomial form
+    - Maintains symbolic conditions for factor validity
+    """
+
+    if not eqs:
+        return []
+
+    exprs = [sympify(eq) for eq in eqs]
+    exprs = [expr.simplify() for expr in exprs]
+
+    if all(expr.is_zero for expr in exprs):
+        return [[]]
+
+    exprs = [expr for expr in exprs if not expr.is_zero]
+
+    if not exprs:
+        return [[]]
+
+    if not gens:
+        gens = set().union(*[expr.free_symbols for expr in exprs])
```

Comment:
This should not assume that the generators will be symbols: `free_symbols` should not be used:
```python
In [18]: Poly(cos(x)**2 - sin(x)**2).factor_list()
Out[18]: 
(1, [(Poly((cos(x)) - (sin(x)), cos(x), sin(x), domain='ZZ'), 1), (Poly((cos(x)) + (sin(x)), cos(x) ↪

↪ , sin(x), domain='ZZ'), 1)])
```

Reply:
Tests should be added for this case where the generators are not symbols.

Reply:
Something like the following should work:

```python 
def _process_system_equations(eqs, gens=None, **kwargs):
    """ Helper function to process system equations and convert to polynomials. """
    
    if not eqs
        return None, []

    exprs = [sympify(eq) for eq in eqs]
    exprs = [expr.simplify() for expr in exprs]

    if all(expr.is_zero for expr in exprs):
        return None, [[]]

    exprs = [expr for expr in exprs if not expr.is_zero]

    if not exprs:
        return None, [[]]

    if not gens:
        try:
            # Let SymPy try to figure out the generators
            polys, opt = parallel_poly_from_expr(exprs, **kwargs)
            gens = opt.gens
        except (GeneratorsNeeded, PolificationFailed):
            # If that fails, check if I was dealing with constants this whole time
            if all(expr.free_symbols == set() for expr in exprs):
                if any(not expr.is_zero for expr in exprs):
                    return None, []
                return None, [[]]
            raise ValueError("Could not determine generators. Please specify them explicitly.")

    # Now we have gens finally, so try polyfication
    try:
        polys, _ = parallel_poly_from_expr(exprs, gens, **kwargs)
    except PolificationFailed as e:
        if all(expr.free_symbols == set() for expr in exprs):
            if any(not expr.is_zero for expr in exprs):
                return None, []
            return None, [[]]
        raise ValueError("The input expressions could not be converted to polynomials.")

    if not polys:
        return None, []

    if any(poly.total_degree() == 0 and not poly.is_zero for poly in polys):
        return None, []

    return polys, None


def factor_system_cond(eqs, gens=None, **kwargs):

    polys, result = _process_system_equations(eqs, gens, **kwargs)
    if result is not None:
        return result

    poly_results = factor_system_poly(polys, **kwargs)

    results = []
    for combo in poly_results:
        factor_conditions = {}
        for factor, conditions in combo:
            expr = factor.as_expr()
            if expr not in factor_conditions:
                factor_conditions[expr] = []
            factor_conditions[expr].extend(conditions)

        boolean_combo = []
        for factor, conditions in factor_conditions.items():
            if not conditions:
                boolean_combo.append((factor, sympy.true))
            else:
                bool_cond = Or(*[Ne(cond.as_expr(), 0) for cond in conditions])
                boolean_combo.append((factor, bool_cond))

        results.append(boolean_combo)

    return results


def factor_system(eqs, gens=None, **kwargs):

    polys, result = _process_system_equations(eqs, gens, **kwargs)
    if result is not None:
        return result

    poly_results = factor_system_poly(polys, **kwargs)
    return [{factor.as_expr() for factor, _ in combo} for combo in poly_results]